corehq/apps/userreports/sql/adapter.py

```diff
@@ -220,7 +220,8 @@ def get_indicator_table(indicator_config, custom_metadata=None):
             _assert = soft_assert('{}@{}'.format('jemord', 'dimagi.com'))
             _assert(False, "Invalid index specified on {}".format(table_name))
             break
-    columns_and_indices = sql_columns + extra_indices
+    constraints = [PrimaryKeyConstraint(*indicator_config.pk_columns)]
```

Comment:
You'll also have to set `is_primary_key=False` for the columns right?

Reply:
For which columns? This can coexist `primary_key=True` arguments in column definitions although supersedes them. It will issue a warning if the columns added here do not match those given that attribute. I had originally removed those attributes entirely, but because of the way we instantiate `sqlalchemy.Table` objects on existing tables, it was erroring when I did it that way. It thought I was trying to remove those columns from the primary key constraint.

Reply:
Re-reading this. Does this only apply to tables that exist? The [SQLAlchemy docs](https://docs.sqlalchemy.org/en/latest/core/constraints.html#sqlalchemy.schema.PrimaryKeyConstraint) warn against specifying it both at the column level and with a `PrimaryKeyConstraint`

Reply:
@calellowitz bump

Reply:
Haven't had a chance to look since you made that comment but will hopefully get some time tomorrow.

Reply:
@calellowitz bump again

Reply:
I was doing to testing with this locally and I found that I had to set `primary_key=True` on all columns specified otherwise I'd get an error when reloading the table:
`
sqlalchemy.exc.ArgumentError: Trying to redefine primary-key column 'owner' as a non-primary-key column on table 'config_report_user-reports_sample_7554612a'
`
I'll make a suggested edit to your test case

Reply:
Actually I see you set `is_primary_key=True` in the datasource config. That's a better idea and I think we should require that to be set.

Reply:
The error you pasted is the reason I used the `primary_key` property in addition to the constraint. It's the one i referred to in that first comment. I spent all day yesterday looking into ways around this, and narrowed it down to this bit of code in sqlalchemy https://github.com/sqlalchemy/sqlalchemy/blob/master/lib/sqlalchemy/sql/schema.py#L1475-L1483, but based on that did not see a way to avoid setting that too.

> Actually I see you set is_primary_key=True in the datasource config. That's a better idea and I think we should require that to be set.

I can change that to be a requirement easily

Reply:
> I can change that to be a requirement easily
:+1: